build.gradle

```diff
@@ -19,12 +19,16 @@ def createLicenseWith = { File license, String startComment, String endComment,
     def tmp = File.createTempFile('tmp', '.tmp')
     tmp.withWriter {
         def w = it
-        w.writeLine(startComment)
+        if (startComment.length() > 0 || !isPython) {
+            w.writeLine(startComment)
+        }
         license.eachLine {
             w << lineComment
             w.writeLine(it)
         }
-        w.writeLine(endComment)
+        if (endComment.length() > 0 || !isPython) {
```

Comment:
for my own undersanding: what is the purpose of these if conditions? might be misunderstanding but it seems like the end state is this will always write for java and not for python, which is the equivalent to what is being set in the start and end comments for either language. 

`triimTrailingWhitespace()` by itself is not enough to get rid of the spaces?

Reply:
In java, we don't have a second linter/styling tool other than spotless so it doesn't really matter if we write a newline or not for startComment/endComment.

However, in python for black, writing an extra newline is actually a big deal and would be reformatted (removed)
So in python, if you don't use a startComment/endComment syntax, then those lines shouldn't be written

Reply:
Thanks for the explanation!

Makes sense. Can we drop a comment capturing this:
```
In java, we don't have a second linter/styling tool other than spotless so it doesn't really matter if we write a newline or not for startComment/endComment.

However, in python, we are using black that double-checks and reformats the code.
Thus, writing an extra empty newline (not removed by trimTrailingWhitespace() is actually a big deal and would be reformatted (removed) because in black's specs, there should be exactly one (or two?) empty lines between license header and imports etc
```
I'm pretty sure I would not remember this after a month :P 

Suggestion - maybe simplify the if conditions to just:
```
if (!isPython) {
            w.writeLine(startComment)
        }
        license.eachLine {
            w << lineComment
            w.writeLine(it)
        }
        if (!isPython) {
            w.writeLine(endComment)
        }
```
since the python relevant logic is already centralised in `createPythonLicenseWith`. I can see argue for checking either parameter, so leave this to your judgement.